app/models/cart.rb

```diff
@@ -1,5 +1,6 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
+  has_many :products, -> {where enabled: true}, through: :line_items
```

Comment:
Lets create 2 associations.
One to get all associated products and another to get enabled products